test/framework/src/main/java/org/elasticsearch/plugins/MockPluginsService.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.Version;
+import org.elasticsearch.action.admin.cluster.node.info.PluginsAndModules;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class MockPluginsService extends PluginsService {
+
+    private static final Logger logger = LogManager.getLogger(MockPluginsService.class);
+
+    private final List<LoadedPlugin> classpathPlugins;
+
+    /**
+     * Constructs a new PluginService
+     *
+     * @param settings         The settings of the system
+     * @param environment      The environment for the plugin
+     * @param classpathPlugins Plugins that exist in the classpath which should be loaded
+     */
+    public MockPluginsService(Settings settings, Environment environment, Collection<Class<? extends Plugin>> classpathPlugins) {
+        super(settings, environment.configFile(), environment.modulesFile(), environment.pluginsFile());
+
+        final Path configPath = environment.configFile();
+
+        List<LoadedPlugin> pluginsLoaded = new ArrayList<>();
+
+        for (Class<? extends Plugin> pluginClass : classpathPlugins) {
+            Plugin plugin = loadPlugin(pluginClass, settings, configPath);
+            PluginInfo pluginInfo = new PluginInfo(
+                pluginClass.getName(),
+                "classpath plugin",
+                "NA",
+                Version.CURRENT,
+                "1.8",
```

Comment:
nit: we should set this to current java version?

Reply:
I'm going to use `JvmInfo.jvmInfo().version()`, but let me know if there's a better way.

Reply:
Yeah there is a simpler way: `Runtime.version().feature()` should work.